watsononlinestore/watson_online_store.py

```diff
@@ -242,13 +243,100 @@ def get_watson_response(self, message):
         return response
 
     @staticmethod
-    def format_discovery_response(response):
-        """Specific to ibm_store_html data
-        """
+    def format_discovery_response(response, data_source):
+        """Format results based on discovery data source."""
         output = []
         if not ('results' in response and response['results']):
             return output
 
+        def get_product_name(entry, data_source):
+            """ Pull product name from entry data for nice user display.
+
+            param: string data_source The name of the discovery data source
+            """
+            product_name = ""
+
+            if data_source == "amazon":
+                # Pull out product name from enriched metadata.
+                if 'extracted_metadata' in entry:
+                    metadata = entry['extracted_metadata']
+                    if 'title' in metadata:
+                        product_name = metadata['title']
+            else:
```

Comment:
We might want to construct it so that it's something like: 
if "amazon":
   # deal with amazon data
elif "ibm_store":
   #deal with ibm_store
#elif <your_dev_stuff>:
   #deal with your custom dev stuff